site/components/pages/Pledge/lib/verifyGnosisSafeMultisig.ts

```diff
@@ -30,3 +31,52 @@ export const verifyGnosisSafeMultisig = async (params: {
 
   await waitForSignedEvent();
 };
+
+// https://github.com/safe-global/safe-contracts/blob/ee92957307653ae6cf7312bbcb1a13c6884ea6ea/src/utils/execution.ts
+export const EIP712_SAFE_MESSAGE_TYPE = {
+  // "SafeMessage(bytes message)"
+  SafeMessage: [{ type: "bytes", name: "message" }],
+};
+
+// https://github.com/safe-global/safe-contracts/blob/ee92957307653ae6cf7312bbcb1a13c6884ea6ea/src/utils/execution.ts
+export const calculateSafeMessageHash = (
+  safe: Contract,
+  message: string,
+  chainId: BigNumberish
+): string => {
+  return utils._TypedDataEncoder.hash(
+    { verifyingContract: safe.address, chainId },
+    EIP712_SAFE_MESSAGE_TYPE,
+    { message: ethers.utils.hashMessage(message) }
+  );
+};
+
+/**
+ * Check if given message string has been signed by the multisig
+ * Throws if no signature was found
+ */
+export const verifyGnosisSignature = async (params: {
+  /** Plain un-hashed string with expected nonce */
+  message: string;
+  /** Address of the multisig contract */
+  address: string;
+}) => {
+  const provider = getJsonRpcProvider();
+  const Safe = new ethers.Contract(
+    params.address,
+    GnosisSafeSignMessageLib.abi,
+    provider
+  );
+  const safeMessageHash = calculateSafeMessageHash(
```

Comment:
Could this be simplified to calling `Safe.getMessageHash` (since we're already importing that function via the ABI).

Or is there a benefit to the approach here?